src/main/java/seedu/address/model/project/Project.java

```diff
@@ -22,9 +22,10 @@ public class Project {
     public static final String VALIDATION_REGEX = "[^\\s].*";
 
     public final String name;
-    public final Deadline deadline;
     public final UniqueEmployeeList employeeList;
+    public final Deadline deadline;
     private final ProjectPriority projectPriority;
+    private final CompletionStatus completionStatus;
```

Comment:
Do you think we could have just implemented this as a boolean field to keep things simpler? Your implementation is very comprehensive though.

Reply:
@timleow `CompletionStatus` is an attribute of `Project`, and by OOP standards, it has a class by itself. If we just did a boolean field "to keep things simple", then it is bad design. Take a look at the `Employee` model, this is how it's attributes are listed:

```Java
public class Employee {

    // Identity fields
    private final Name name;
    private final Phone phone;
    private final Email email;

    // Data fields
    private final Address address;
    private final Project project;
    private final Set<Tag> tags = new HashSet<>();

   ...
}
```
Every attribute here is a class.